src/Host/Host.php

```diff
@@ -202,31 +192,33 @@ public function multiplexing(bool $multiplexing = true)
         return $this;
     }
 
-    /**
-     * @return array
-     */
-    public function getOptions()
+    public function getSshArguments() : Arguments
     {
-        return $this->options;
+        $this->initOptions();
+        return $this->sshArguments;
     }
 
-    /**
-     * @param array $options
-     * @return $this
-     */
-    public function options(array $options)
+    public function sshOptions(array $options) : Host
     {
-        $this->options = $options;
+        $this->sshArguments = $this->sshArguments->withOptions($options);
         return $this;
     }
 
-    /**
-     * @param string $option
-     * @return $this
-     */
-    public function addOption(string $option)
+    public function sshFlags(array $flags) : Host
+    {
+        $this->sshArguments = $this->sshArguments->withFlags($flags);
+        return $this;
+    }
+
+    public function addSshOption(string $option, $value) : Host
+    {
+        $this->sshArguments = $this->sshArguments->withOption($option, $value);
```

Comment:
My idea behind this was to keep the arguments for the host from being altered accidentally outside of the class. E.g if a task called getSshArguments then added more flags / options without immutability this would change the sshArguments on the host object. 

If you don't think this is something we need to be concerned with it's a relatively simple change I can do.